### PR TITLE
Automatically delete expired records from the partial table

### PR DESCRIPTION
**Pull request by pchacker**
Add the function of automatically deleting expired records of “partial” table, which can reduce the disk occupation of sqlite.

### CODE REVIEW

File: pool/store/sqlite_store.py [hunk 1]
```diff
@@ -167,7 +167,12 @@ async def clear_farmer_points(self) -> None:
         cursor = await self.connection.execute(f"UPDATE farmer set points=0")
         await cursor.close()
         await self.connection.commit()
-
+        
+    async def auto_delete_partial(self,retetion_time: uint64) -> None:
+        cursor = await self.connection.execute(f"DELETE from partial where timestamp <?", (retetion_time,))
```
Comment (mariano54) [inline pool/store/sqlite_store.py:172]: The timestamp on the table is a UNIX time,. so this is not correct

**Review by mariano54 — CHANGES_REQUESTED** [review]

File: pool/store/sqlite_store.py (hunk 1 above)
Comment (pchacker) [reply pool/store/sqlite_store.py:172]: await self.store.auto_delete_partial(time.time() - self.partial_retetion_time)

retetion_time = time.time() - pool_config["partial_retetion_time"]

DELETE from partial where timestamp < retetion_time 
It means delete the data before (current time - pool_config["partial_retetion_time"]).

**Comment by pchacker** [comment]
Important bug fixes #223 

farmer request pool "post_farmer", "put_farmer" RPC interface, causing points to be replace by historical data.

File: pool/store/sqlite_store.py [hunk 2]
```diff
@@ -71,25 +71,50 @@ def _row_to_farmer_record(row) -> FarmerRecord:
...
+	             farmer_record.points,
+	             farmer_record.difficulty,
+	             farmer_record.payout_instructions,
+	             int(farmer_record.is_pool_member),
+	         ),
+            )
+            #update for Exist
+        else:
+            cursor = await self.connection.execute(
+	         f"UPDATE farmer SET p2_singleton_puzzle_hash=?, delay_time=?, delay_puzzle_hash=?, authentication_public_key=?, singleton_tip=?, singleton_tip_state=?, payout_instructions=?, is_pool_member=? WHERE launcher_id=?",
```
Comment (jianghushinian) [inline pool/store/sqlite_store.py:101]: According to my observation, this project specification should be less than 120 characters per line of code.

File: pool/store/sqlite_store.py [hunk 3]
```diff
@@ -71,25 +71,50 @@ def _row_to_farmer_record(row) -> FarmerRecord:
         )
 
     async def add_farmer_record(self, farmer_record: FarmerRecord, metadata: RequestMetadata):
+        #Find the launcher_id exists.
```
Comment (jianghushinian) [inline pool/store/sqlite_store.py:74]: There should be a space for the # and comment. refer to pep8 specification.

**Pull request closed without merging**